assets/js/components/settings/settings-module.js

```diff
@@ -91,26 +92,18 @@ class SettingsModule extends Component {
 
 	async activateOrDeactivate() {
 		try {
-			const { active } = this.state;
-			const newActiveState = ! active;
-
 			this.setState( { isSaving: true } );
 
 			await activateOrDeactivateModule(
 				data,
-				this.props.slug,
-				newActiveState
+				this.props.slug
 			);
 
 			await refreshAuthentication();
-
-			if ( false === newActiveState ) {
-				data.invalidateCacheGroup( TYPE_MODULES, this.props.slug );
-			}
+			data.invalidateCacheGroup( TYPE_MODULES, this.props.slug );
 
 			this.setState( {
 				isSaving: false,
-				active: newActiveState,
```

Comment:
Can we restore this method to the way it was? It's confusing since this is only ever used to deactivate the module (rather than also activate as it says) but we should limit changes in this file to only what is necessary as it has its own issue to refactor it. Less changes makes this easier to review as well.